Libraries/LibJS/Runtime/PrimitiveString.h

```diff
@@ -47,26 +47,50 @@ class PrimitiveString final : public Cell {
 
     ThrowCompletionOr<Optional<Value>> get(VM&, PropertyKey const&) const;
 
-private:
-    explicit PrimitiveString(PrimitiveString&, PrimitiveString&);
-    explicit PrimitiveString(String);
-    explicit PrimitiveString(Utf16String);
+protected:
+    enum class RopeTag { Rope };
```

Comment:
Not sure what is nicer, but alternatively one could use a
```c++
struct RopeTag {
    explicit constexpr RopeTag() = default
};
```
akin to `OptionalNone`
like

Reply:
FWIW, my implementation used a `Badge` to achieve this. All of these options do pretty much the same thing :shrug:

Reply:
That actually sounds nicer, I'd say